triton-vm/src/table/master_table.rs

```diff
@@ -284,35 +284,69 @@ where
         let evaluation_domain = self.evaluation_domain();
         let randomized_trace_domain = self.randomized_trace_domain();
         let num_rows = evaluation_domain.length;
+
+        profiler!(start "polynomial zero-initialization");
         let mut interpolation_polynomials = Array1::zeros(Self::NUM_COLUMNS);
+        profiler!(stop "polynomial zero-initialization");
 
         // compute interpolants
+        profiler!(start "interpolation");
         Zip::from(self.randomized_trace_table().axis_iter(Axis(1)))
             .and(interpolation_polynomials.axis_iter_mut(Axis(0)))
             .par_for_each(|trace_column, poly| {
                 let trace_column = trace_column.as_slice().unwrap();
                 let interpolation_polynomial = randomized_trace_domain.interpolate(trace_column);
                 Array0::from_elem((), interpolation_polynomial).move_into(poly);
             });
+        profiler!(stop "interpolation");
 
-        let mut extended_trace = Vec::<FF>::new();
+        let mut extended_trace = Vec::<FF>::with_capacity(0);
         let num_elements = num_rows * Self::NUM_COLUMNS;
-        let should_cache = crate::config::cache_lde_trace().map_or_else(
-            || extended_trace.try_reserve_exact(num_elements).is_ok(),
-            |cd| cd == CacheDecision::Cache,
-        );
+        let should_cache = match crate::config::cache_lde_trace() {
+            Some(CacheDecision::Cache) => {
+                extended_trace = Vec::with_capacity(num_elements);
+                true
+            }
+            None => extended_trace.try_reserve_exact(num_elements).is_ok(),
+            Some(CacheDecision::NoCache) => false,
+        };
 
         if should_cache {
-            extended_trace.resize(num_elements, FF::zero());
+            profiler!(start "resize");
```

Comment:
I'd like to see an `assert!(extended_trace.capacity() >= num_elements);` here. This will convince ourselves of the safety of below `unsafe` code and will detect regressions of this kind.

Note that “greater than or equal” is the correct check, since `Vec::with_capacity` “is allowed to allocate for more elements than `capacity`” ([source](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity)).